api/novelist.py

```diff
@@ -433,6 +460,73 @@ def get_recommendations(self, metadata, recommendations_info):
                 metadata.recommendations += self._extract_isbns(book_info)
         return metadata
 
+    def get_isbns_from_query(self, library):
+        collectionList = []
+        for c in library.collections:
+            collectionList.append(c.id)
+
+        LEFT_OUTER_JOIN = True
+        i1 = aliased(Identifier)
+        i2 = aliased(Identifier)
+        e = aliased(Equivalency)
+        lp = aliased(LicensePool)
+
+        isbnQuery = select(
+            [i1.identifier, i1.type, i2.identifier]
+        ).select_from(
+            join(lp, i1, i1.id==lp.identifier_id)
+            .join(e, i1.id==e.input_id, LEFT_OUTER_JOIN)
+            .join(i2, e.output_id==i2.id, LEFT_OUTER_JOIN)
+        ).where(
+            and_(
+                lp.collection_id.in_(collectionList),
+                or_(i1.type=="ISBN", i2.type=="ISBN")
+            )
+        ).alias('lp_isbns')
+
+        result = self._db.execute(isbnQuery)
+
+        isbns = []
+        for res in result:
+            if (res[1] == "ISBN"):
+                isbns.append(res[0])
+            else:
```

Comment:
This is a minor point but I think this ought to be `elif res[2] is not null` to handle the case where an ISBN is marked as equivalent to another ISBN.